source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -0,0 +1,135 @@
+    if (!(condition)) {                                                                            \
+    } else                                                                                         \
+      logstream
+
+#define QUIC_LOG_IF_IMPL(severity, condition)                                                      \
+  QUIC_LOG_IMPL_INTERNAL((condition) && quic::IsLogLevelEnabled(quic::severity),                   \
+                         quic::QuicLogEmitter(quic::severity).stream())
+
+#define QUIC_LOG_IMPL(severity) QUIC_LOG_IF_IMPL(severity, true)
+
+#define QUIC_VLOG_IF_IMPL(verbosity, condition)                                                    \
+  QUIC_LOG_IMPL_INTERNAL((condition) && quic::IsVerboseLogEnabled(verbosity),                      \
+                         quic::QuicLogEmitter(quic::INFO).stream())
+
+#define QUIC_VLOG_IMPL(verbosity) QUIC_VLOG_IF_IMPL(verbosity, true)
+
+// TODO(wub): Implement QUIC_LOG_FIRST_N_IMPL.
+#define QUIC_LOG_FIRST_N_IMPL(severity, n) QUIC_LOG_IMPL(severity)
+
+// TODO(wub): Implement QUIC_LOG_EVERY_N_SEC_IMPL.
+#define QUIC_LOG_EVERY_N_SEC_IMPL(severity, seconds) QUIC_LOG_IMPL(severity)
+
+#define QUIC_PLOG_IMPL(severity)                                                                   \
+  QUIC_LOG_IMPL_INTERNAL(quic::IsLogLevelEnabled(quic::severity),                                  \
+                         quic::QuicLogEmitter(quic::severity).SetPerror().stream())
+
+#define QUIC_LOG_INFO_IS_ON_IMPL quic::IsLogLevelEnabled(quic::INFO)
+#define QUIC_LOG_WARNING_IS_ON_IMPL quic::IsLogLevelEnabled(quic::WARNING)
+#define QUIC_LOG_ERROR_IS_ON_IMPL quic::IsLogLevelEnabled(quic::ERROR)
+
+#ifdef NDEBUG
+// Release build
+#define QUIC_COMPILED_OUT_LOG() QUIC_LOG_IMPL_INTERNAL(false, quic::NullLogStream().stream())
+#define QUIC_DVLOG_IMPL(verbosity) QUIC_COMPILED_OUT_LOG()
+#define QUIC_DVLOG_IF_IMPL(verbosity, condition) QUIC_COMPILED_OUT_LOG()
+#define QUIC_DLOG_IMPL(severity) QUIC_COMPILED_OUT_LOG()
+#define QUIC_DLOG_IF_IMPL(severity, condition) QUIC_COMPILED_OUT_LOG()
+#define QUIC_DLOG_INFO_IS_ON_IMPL 0
+#define QUIC_NOTREACHED_IMPL()
+#else
+// Debug build
+#define QUIC_DVLOG_IMPL(verbosity) QUIC_VLOG_IMPL(verbosity)
+#define QUIC_DVLOG_IF_IMPL(verbosity, condition) QUIC_VLOG_IF_IMPL(verbosity, condition)
+#define QUIC_DLOG_IMPL(severity) QUIC_LOG_IMPL(severity)
+#define QUIC_DLOG_IF_IMPL(severity, condition) QUIC_LOG_IF_IMPL(severity, condition)
+#define QUIC_DLOG_INFO_IS_ON_IMPL QUIC_LOG_INFO_IS_ON_IMPL
+#define QUIC_NOTREACHED_IMPL() NOT_REACHED_GCOVR_EXCL_LINE
+#endif
+
+#define QUIC_PREDICT_FALSE_IMPL(x) ABSL_PREDICT_FALSE(x)
+
+namespace quic {
+
+using QuicLogLevel = spdlog::level::level_enum;
+
+static const QuicLogLevel INFO = spdlog::level::info;
+static const QuicLogLevel WARNING = spdlog::level::warn;
+static const QuicLogLevel ERROR = spdlog::level::err;
+static const QuicLogLevel FATAL = spdlog::level::critical;
```

Comment:
Just something to keep in mid as we code, in Envoy only critical logs flush immediately.
It won't cause problems and I think while we could set Enovy QUIC logs to flush at error it makes more sense to stick with the defaults, but worth being aware when we debug.